src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -785,8 +824,22 @@ VideoDecoder::DecodedOutput VideoDecoder::convertAVFrameToDecodedOutput(
   output.durationSeconds = ptsToSeconds(
       getDuration(frame), formatContext_->streams[streamIndex]->time_base);
   if (output.streamType == AVMEDIA_TYPE_VIDEO) {
-    output.frame =
-        convertFrameToTensorUsingFilterGraph(streamIndex, frame.get());
+    if (streamInfo.colorConversionLibrary == ColorConversionLibrary::SWSCALE) {
```

Comment:
Nit: when checking enums, I prefer explicit value checks on all branches:
```
if (val == Kind::A) {
  handle_a(val);
} else if (val == Kind::B) {
  handle_b(val);
} else {
  throw exception("Unhandled Kind: " + to_string(val));
}
```
The reason is that if we ever change the enum values, then we'll fail loudly if there is any place we did not handle it. If we handle a case by default in a bare-else branch, we can easily miss this.

Reply:
Done